robot/Cumulus/tests/browser/gift_entry/test_batch_status_adding_removing_gifts.robot

```diff
@@ -0,0 +1,141 @@
+    Click Gift Entry Button          Next
+    Click Gift Entry Button          Save
+    Current Page Should Be           Form                          Gift Entry
+
+    ${BATCH_ID} =                    Save Current Record ID For Deletion     ${NS}DataImportBatch__c
+
+    Fill Gift Entry Form
+    ...                              Donor Type=Contact1
+    ...                              Existing Donor Contact=${CONTACT}[Name]
+    ...                              Donation Amount=${AMOUNT}
+    ...                              Donation Date=Today
+
+    Click Special Button             Save & Enter New Gift
+    Verify Table Field Values        Batch Gifts
+    ...                              Status=Dry Run - Validated
+
+    # Update the gift value and check the value gets  reflected in the table
+    Perform Action On Datatable Row   ${CONTACT}[Name]          Open
+    Fill Gift Entry Form
+     ...                              Donation Amount=${AMOUNT_UPDATED}
+    Click Special Button              Update
+    Verify Table Field Values         Batch Gifts
+     ...                              Donation Amount=$${AMOUNT_UPDATED}.00
+    # Run and validate the Batch data import status
+
+    Process And Validate Batch       Completed
+    Verify Table Field Values        Batch Gifts
+    ...                              Status=Imported
+    # Verify the backend dataimport batch job using the API call
+    Verify Expected Values    nonns  DataImportBatch__c    ${BATCH_ID}
+    ...                              Batch_Status__c=Completed
+
+    #Scroll to the top of the page, add gift for contact associated with org account
+    Scroll Page To Location          0      0
+    Fill Gift Entry Form
+    ...                              Donor Type=Contact1
+    ...                              Existing Donor Contact=${CONTACT2}[Name]
+    ...                              Donation Amount=${AMOUNT}
+    ...                              Donation Date=Today
+
+    Click Special Button             Save & Enter New Gift
+    Verify Table Field Values        Batch Gifts
+    ...                              Status=Dry Run - Error
+    Process And Validate Batch       Errors
+    Verify Table Field Values        Batch Gifts
+    ...                              Status=Failed
+
+    # Verify the backend dataimport batch job
+    Verify Expected Values    nonns  DataImportBatch__c    ${BATCH_ID}
+    ...                              Batch_Status__c=Failed - Needs Review
+
+    #Delete the row that has Dry run status and trigger the batch job to see it completed
+    Perform Action On Datatable Row  ${CONTACT2}[Name]          Delete
+
+    # Verify the backend dataimport batch job
+    Verify Expected Values    nonns  DataImportBatch__c    ${BATCH_ID}
+    ...                              Batch_Status__c=Completed
+
+    Process And Validate Batch       Completed
```

Comment:
since there are no new records, do u still need to process batch?

Reply:
I do not need this one